### ISSUE TITLE
I wonder how to implement spatial Variant reconstruction loss on irregular masks

### ISSUE THREAD

**Issue by SunHu1** [shepnerd/inpainting_gmcnn#14]
Its easy to implement spatial Variant reconstruction loss on  the rectangle masks.
Did you implement spatial Variant reconstruction loss on irregular masks?

**Comment by shepnerd** [comment]
We did not presume the mask type for spatial variant mask weights. The given implementations (priority_loss_mask in inpaint_ops.py [tensorflow] or ConfidenceDrivenMaskLayer in layer.py [pytorch]) can work both on regular or irregular mask. The idea of this spatial variant mask weight (or confidence-driven mask) is to propagate the confidence of the known pixels to the unknown ones by Gaussian smoothing in an iterative manner. The difference of using this on regular and irregular masks is to apply different setups of the parameters.

**Comment by SunHu1** [comment]
Thank you for your reply,I got it.
